### PR TITLE
Move accessibility toggle and instructions from body to navigation

### LINKED ISSUE #1182: Add accessibility toggle button to navigation bar
- To allow touch devices to toggle accessibility fully rather than having it always on.
- Must be the first tabbable item for desktops
- Need to add an appropriate icon to the vanilla fonts

### PR DESCRIPTION
**Pull request by chris-steele**
#1182 
https://github.com/adaptlearning/adapt_framework/issues/1599

Concerns accessibility toggle button and instructions

### Changes
- Moved toggle and instructions to navigation bar
- No more tab capture
- No more forcing active on touch devices
- Added standard orange border to toggle button so clear to keyboard users
- `_isEnabledOnTouchDevices` to be removed from `config.json`

### CODE REVIEW

**Comment by oliverfoster** [comment]
this is beautiful... i think i almost started crying just.

**Comment by oliverfoster** [comment]
In accessibility.js can you add something like:

```javascript
        initialize: function() {

            //RUN ONCE
            if (this._isLoaded) return;


            this.compatibility();

        },

        compatibility: function() {
            var $legacyElements = $("body > #accessibility-toggle, body > #accessibility-instructions");
            if ($legacyElements.length) {
                Adapt.log.warn("DEPRECATED - #accessibility-toggle and #accessibility-instructions have been moved to the navigation bar. Please remove them from all of your .html files.");
                $legacyElements.remove();
            }
        },
```

Some people have bespoke versions of spoor, which we need to not break. the existence of those elements breaks accessibilityView.js el selector. So this or something similar needs to go in for the moment.

**Comment by oliverfoster** [comment]
I dunno if you want to do that in this one or a later pr on a separate ticket. The "legacy" stuff can go from accessibility.js, ```_legacyFocusElements, _tabIndexElements, setupLegacyFocusClasser, removeLegacyFocusClasser, setupLegacy, revertLegacy, onElementFocused, onElementBlurred``` those are cos ie8 doesn't have pseudo classes for :focus and :blur.

**Comment by oliverfoster** [comment]
Otherwise, total +1, works a treat. Thank you.

File: src/core/js/accessibility.js [hunk 1]
```diff
@@ -1,27 +1,20 @@
 define([
     'core/js/adapt',
-    'core/js/views/accessibilityView',
     'a11y'
-], function(Adapt, AccessibilityView) {
+], function(Adapt) {
```
Comment (oliverfoster) [inline src/core/js/accessibility.js]: -

**Comment by chris-steele** [comment]
> I dunno if you want to do this in this pr or a later pr on a separate ticket. The "legacy" stuff can go from accessibility.js

Yep I was thinking there are enough changes in this PR so would do this separately.

**Comment by chris-steele** [comment]
We'll need a framework version bump for this and an update to the bower.json for the language picker and spoor plugins.

**Comment by moloko** [comment]
ah... https://github.com/adaptlearning/adapt_framework/pull/1624

**Comment by oliverfoster** [comment]
1624 still needs device.js reverting :/ lordy...

File: src/core/templates/navigation.hbs [hunk 2]
```diff
@@ -1,7 +1,12 @@
 {{! make the _globals object in course.json available to this template}}
 {{import_globals}}
 {{! All links in the nav bar should have an attribute of data-event, navigationView uses this to fire a global event}}
-<div class="navigation-inner clearfix" role="navigation" aria-label="{{_globals._accessibility._ariaLabels.navigation}}">
+<div class="navigation-inner clearfix" role="navigation">
+	{{#if _accessibility._isEnabled}}
+		<button id="accessibility-toggle" class="base icon icon-accessibility a11y-ignore-focus"></button>
+		<span id="accessibility-instructions" class="aria-label" role="region" tabindex="0"></span>
+	{{/if}}
+	{{{a11y_aria_label _globals._accessibility._ariaLabels.navigation}}}
```
Comment (ryan-lynch) [inline src/core/templates/navigation.hbs]: Minor, but think the indentation is a bit off here.

**Comment by ryan-lynch** [comment]
Looks good to me +1

File: src/core/templates/navigation.hbs (hunk 2 above)
Comment (chris-steele) [reply src/core/templates/navigation.hbs]: Tabs crept in - have converted to spaces

**Comment by oliverfoster** [comment]
awaiting https://github.com/adaptlearning/adapt_framework/issues/1831
and https://github.com/adaptlearning/adapt_framework/issues/1829
please do not merge until the language picker code is ready

**Comment by oliverfoster** [comment]
this pr may help with > https://github.com/adaptlearning/adapt_framework/issues/1752

File: src/core/js/accessibility.js [hunk 3]
```diff
@@ -59,41 +48,38 @@ define([
             this.listenTo(Adapt, 'pageView:ready menuView:ready router:plugin', this.onNavigationEnd);
         },
 
+        removeLegacyElements: function() {
+            var $legacyElements = $("body > #accessibility-toggle, body > #accessibility-instructions");
```
Comment (tomgreenfield) [inline src/core/js/accessibility.js]: Would you mind switching to the more concise and efficient `$.children()`?
```js
$("body").children("#accessibility-toggle, #accessibility-instructions");
```
And switching the logic to return early?

File: src/core/js/accessibility.js [hunk 4]
```diff
@@ -248,64 +219,25 @@ define([
...
             //MAKE FOCUS RIGHT
-            this._isButtonRedirectionOn = true;
             _.delay(_.bind(function() {
                 this.focusInitial();
             }, this), 500);
 
         },
 
         setNavigationBar: function() {
+            var navArias = $(".navigation .aria-label").not('#accessibility-instructions');
```
Comment (tomgreenfield) [inline src/core/js/accessibility.js]: `$` prefix on variable?

File: src/core/js/views/navigationView.js [hunk 5]
```diff
@@ -1,13 +1,15 @@
...
+    'core/js/views/accessibilityView'
+], function(Adapt, AccessibilityView) {
 
     var NavigationView = Backbone.View.extend({
 
         className: "navigation",
 
         initialize: function() {
             this.listenToOnce(Adapt, 'courseModel:dataLoading', this.remove);
+            this.listenToOnce(Adapt, 'accessibility:toggle', this.onA11yToggle);
```
Comment (tomgreenfield) [inline src/core/js/views/navigationView.js]: Could you switch to object syntax here for these two calls?
```js
this.listenToOnce(Adapt, {
    ...
});
```

File: src/core/templates/navigation.hbs [hunk 6]
```diff
@@ -1,7 +1,12 @@
 {{! make the _globals object in course.json available to this template}}
 {{import_globals}}
 {{! All links in the nav bar should have an attribute of data-event, navigationView uses this to fire a global event}}
-<div class="navigation-inner clearfix" role="navigation" aria-label="{{_globals._accessibility._ariaLabels.navigation}}">
+<div class="navigation-inner clearfix" role="navigation">
+    {{#if _accessibility._isEnabled}}
+        <button id="accessibility-toggle" class="base icon icon-accessibility a11y-ignore-focus"></button>
+        <span id="accessibility-instructions" class="aria-label" role="region" tabindex="0"></span>
```
Comment (tomgreenfield) [inline src/core/templates/navigation.hbs]: @oliverfoster, are we using `role`s any more?

**Comment by moloko** [comment]
@chris-steele if you could address @tomgreenfield 's suggestions I think we're done with all this? I gave it a test today and it all looks good to me.

File: src/core/templates/navigation.hbs (hunk 6 above)
Comment (oliverfoster) [reply src/core/templates/navigation.hbs]: yes, roles just can't be nested.

**Comment by chris-steele** [comment]
@tomgreenfield I'm happy for you to make the changes you've raised to get this moving. I'm not sure when I'll get around to it at the moment.

**Pull request merged**